enterprise/frontend/src/metabase-enterprise/model_persistence/components/ModelCacheControl/ModelCacheControl.tsx

```diff
@@ -1,85 +1,65 @@
-import { useCallback, useState } from "react";
 import { t } from "ttag";
 
-import LoadingSpinner from "metabase/components/LoadingSpinner";
-import Button from "metabase/core/components/Button";
-import Databases from "metabase/entities/databases";
-import { delay } from "metabase/lib/promise";
-import { CardApi } from "metabase/services";
+import {
+  skipToken,
+  useGetDatabaseQuery,
+  usePersistModelMutation,
+  useUnpersistModelMutation,
+} from "metabase/api";
+import { DelayedLoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper/DelayedLoadingAndErrorWrapper";
+import { Switch, Tooltip } from "metabase/ui";
 import type Question from "metabase-lib/v1/Question";
-import type Database from "metabase-lib/v1/metadata/Database";
+import type { ModelCacheRefreshStatus } from "metabase-types/api";
 
-import { SpinnerContainer } from "./ModelCacheControl.styled";
-
-interface ModelCacheControlProps {
+export function ModelCacheToggle({
+  persistedModel,
+  model,
+}: {
+  persistedModel?: ModelCacheRefreshStatus;
   model: Question;
-  size?: number;
-  onChange?: (isPersisted: boolean) => void;
-}
-
-type DatabaseEntityLoaderProps = {
-  database?: Database;
-};
+}) {
+  const [persistModel] = usePersistModelMutation();
+  const [unpersistModel] = useUnpersistModelMutation();
+  const { data: database, isLoading: isLoadingDb } = useGetDatabaseQuery(
+    model.databaseId() ? { id: model.databaseId() as number } : skipToken,
+  );
 
-export const toggleModelPersistence = async (
-  model: Question,
-  onChange?: (isPersisted: boolean) => void,
-) => {
-  const id = model.id();
-  const isPersisted = model.isPersisted();
-  try {
-    if (isPersisted) {
-      await CardApi.unpersist({ id });
-    } else {
-      await CardApi.persist({ id });
-    }
-    onChange?.(!isPersisted);
-  } catch (err) {
-    console.warn("Failed to persist/unpersist model");
-  } finally {
-    await delay(200);
+  if (isLoadingDb) {
+    return <DelayedLoadingAndErrorWrapper loading error={null} />;
```

Comment:
Would it be valuable to handle the error here?

Reply:
I actually really intentionaly don't want to handle it there, but that's not at all obvious, so I will add a comment explaining why